### ISSUE TITLE
Heatmap not resizing

### ISSUE THREAD

**Issue by subhajitm1993** [apexcharts/apexcharts.js#2211]
# Bug report
I am using angular apexchart api
I am trying to show a 'confusion matrix' using a heatmap. By default, the series contains 50 elements. The heatmap is thus generated with some height inside the chart property.

this is the config
![default config](https://user-images.githubusercontent.com/56729811/107033414-91ee7f00-67b5-11eb-9bc1-c8d16d55f17b.png)

this is the view
![default view](https://user-images.githubusercontent.com/56729811/107033440-99158d00-67b5-11eb-9fa2-2a8ca41fa77f.JPG)

Then the logic is such that, a user can filter some elements(say 20) and we have to show a heatmap of those many elements. Hence I update the series array. Also, I update the height property. However, the heatmap is not resizing, but keeping the original height.

this is the changed config
![filtered config](https://user-images.githubusercontent.com/56729811/107033466-a0d53180-67b5-11eb-8baa-32fdbadc620e.png)

this is the changed view
![filtered view](https://user-images.githubusercontent.com/56729811/107033490-a763a900-67b5-11eb-9029-e7dc3cc29cfa.JPG)

this view height and width I want to update and control from code

## Explanation

- What is the behavior you expect?
the chart should resize as per the height/width property
- What is happening instead?
the chart is keeping the original height
- What error message are you getting?
no errors logged

**Comment by subhajitm1993** [comment]
public chartOptions: ChartOptions = {
    series: [],
    chart: {
      type: "heatmap",
      redrawOnParentResize: true,
      redrawOnWindowResize: true
    },
    plotOptions: {
      heatmap: {
        radius:0,
        shadeIntensity: 0.5,
        colorScale: {
          ranges: [
            {
              from: 0,
              to: 0,
              name: "None",
              color: "#FFFFFF"
            },
            {
              from: 1,
              to: 1,
              name: "Same",
              color: "#000000"
            },
            {
              from: 2,
              to: 2,
              name: "Low",
              color: "#008000"
            },
            {
              from: 3,
              to: 3,
              name: "Medium",
              color: "#FFBF00"
            },
            {
              from: 4,
              to: 4,
              name: "High",
              color: "#FF0000"
            }
          ]
        }
      }
    },
    dataLabels: {
      enabled: false
    },
    stroke: {
      width: 1,
      colors:["#000000"]
    },
    title: {
      text: "SOD Matrix",
      align: 'center'
    },
    xaxis: {
      type: 'category',
      position: 'bottom',
      labels: {
        show: true,
        rotate: -90,
        maxHeight: 1000,
        rotateAlways: true,
      },
      tooltip: {
        enabled: false
      }
    },
    yaxis: {
      labels: {
        show: true,
        maxWidth: 1000,
        align: 'right'
      }
    },
    tooltip: {
      shared: true,
      intersect: false,
      x: {
        show: true
      },
      y: {
        formatter: function (val) {
          if (val == 0) return "None"
          if (val == 1) return "Same"
          if (val == 2) return "Low"
          if (val == 3) return "Medium"
          if (val == 4) return "High"
        }
      },
      marker: {
        show: true
      },
      followCursor: false,
    },
    legend: {
      show: true,
      position: 'top',
      horizontalAlign: 'center',
      markers: {
        strokeWidth: 1,
        strokeColor: '#000000'
      }
    }

  };



this is the code.
Im pusing in the series and height (based on the number of elements in series) dynamically

**Comment by subhajitm1993** [comment]
I think I found the issue. Angular API was not giving any error messages, however, the normal js API was giving some error messages in the browser console. the problem that I found by debugging was that the height and width supplied in chart options(and not CSS), that is the SVG height and width, was not large enough for everything to be rendered. in my case the labels were very large, so the number SVG height and width was going negative. Hence, giving a sufficiently large height and width rendered the plot.

I am not sure what is the correct way to handle this error in the library, but it would have been nice if there was some way to propagate the error to the users. Not my call anyways

To get an optimal height and width, I borrowed your graphics class.

Also, I started doing using the normal js API, the angular API was still not responding appropriately, as much as I saw.

Thanks again
